rllib/agents/trainer.py

```diff
@@ -282,6 +282,12 @@
     # of currently healthy workers is reported as the "num_healthy_workers"
     # metric.
     "ignore_worker_failures": False,
+    # Whether - upon a worker failure - RLlib will try to recreate the lost worker as
+    # an identical copy of the failed one. The new worker will only differ from the
+    # failed one in its `self.recreated_worker=True` property value. It will have
+    # the same `worker_index` as the original one.
+    # If True, the `ignore_worker_failures` setting will be ignored.
+    "recreate_failed_workers": False,
```

Comment:
do you think we should set this default to True, and deprecate ignore_worker_failures?
can't think of why anyone would not want this.

Reply:
Hmm, yeah, the danger with that is that if there is a real problem with one or more workers (e.g. the env is broken) and they keep crashing, RLlib will hopelessly try to restart them over and over again. This particular problem has not been fixed.
Maybe we can add an extra key: `failed_worker_recreation_num_retries`. ?

Reply:
normally, I'd say it's ok to keep restarting.
it's monitoring's job to alert the users that their job has been consistently restarting.
since there may be a lot of workers, I feel like you need to set the default pretty high anyways, if they are running on spot for example.

Reply:
just to clarify a bit more, I think absolute # is not a good indication. a job that runs 2 weeks may accumulate a lot of restarts, and we essentially want to detect many restarts in a short period of time, which is why it's better handled by monitoring.

Reply:
Yeah, makes sense! Let's do this in a follow-up PR, though.
Then we can also discuss, whether we should soft-deprecate `ignore_worker_failures` or just leave its functionality as-is.

Reply:
I think we should soft-deprecate ignore_worker_failure.